### ISSUE TITLE
Ignore case and whitespace for auth scheme (token-type)

### ISSUE THREAD

**Issue by jstirnaman** [influxdata/influxdb_iox#7659]
# Proposal

IOx (and IOx on IDPE) should be case-insensitive when handling authorization values (scheme-name and token).
IOx (and IOx on IDPE) should accept the authorization value (scheme-name and token) with or without leading spaces.

...as per HTTP Authentication [RFC7235](https://datatracker.ietf.org/doc/html/rfc7235#section-2.1):

>    HTTP provides a simple challenge-response authentication framework
>    that can be used by a server to challenge a client request and by a
>    client to provide authentication information.  It uses a case-
>    insensitive token as a means to identify the authentication scheme,
>    followed by additional information necessary for achieving
>    authentication via that scheme.  The latter can be either a comma-
>    separated list of parameters or a single sequence of characters
>    capable of holding base64-encoded information.

OAuth 2.0 defines the ["Bearer" token-type](https://www.rfc-editor.org/rfc/rfc6750#page-14), but there is no requirement that the capitalization be preserved.

## Why

Developer happiness

# Current behavior:

- Cloud single-tenant `/api/v2/write` accepts sentence-cased auth scheme names (`Bearer`, `Token`) and rejects lowercase auth scheme names (`bearer`, `token`).
- IOx gRPC accepts a sentence-case auth scheme name (`Bearer`) and rejects a lowercase auth scheme name (`bearer`).
- IOx gRPC accepts an auth scheme name with no leading whitespace (`Authorization:Bearer`) and rejects authorization when the scheme name as leading whitespace:

  ```
  ( `influxdb_iox query --host $INFLUX_URL --header "Authorization: Bearer $INFLUX_TOKEN" \)
  ```

  ```
  Error querying: Tonic(Status { code: Unauthenticated, message: "Unauthenticated", metadata: MetadataMap { headers: 
  {"content-type": "application/grpc", "server": "envoy", "x-envoy-upstream-service-time": "44", "content-length": "0", "date": 
  "Tue, 25 Apr 2023 14:09:24 GMT"} }, source: None })
  ```

# Desired behavior:

- Be liberal in accepting input.
- Authorization header values should be case-insensitive; don't enforce case-sensitive auth scheme names (token-type). 
- For HTTP API requests, allow the following:
  - `authorization: <bearer | Bearer | Token | token> <token>`
  - `authorization:<bearer | Bearer | Token | token> <token>`
- For gRPC requests, allow the following:
  -  `authorization: <bearer | Bearer> <token>`
  - `authorization:<bearer | Bearer> <token>`
 
Relevant links:
- https://github.com/influxdata/influxdb_iox/blob/781d6c040d0dd5c66709d9c65847dcacbb995e88/router/src/server/http/write/single_tenant/auth.rs#L15-L16
- https://github.com/influxdata/influxdb_iox/blob/7a1b0b6fc2aec53f23310c6edb4918f8d233a818/service_grpc_flight/src/lib.rs#L773-L774
- https://github.com/hashicorp/consul/issues/6568
- https://datatracker.ietf.org/doc/html/rfc7235#section-2.1
- https://www.rfc-editor.org/rfc/rfc6750#page-14

**Comment by jstirnaman** [comment]
@gunnaraasen Please weigh in. We'd need to submit this to IDPE or Cloud infra as well?

**Comment by gunnaraasen** [comment]
Thanks, @jstirnaman.

I think this flexibility is only really important for the HTTP API, since we want to keep compatibility with the many existing influx HTTP clients. The FlightSQL and gRPC APIs are new, but couldn't hurt to be flexible on what we accept, while documenting only the correct way.

**Comment by jstirnaman** [comment]
> I think this flexibility is only really important for the HTTP API

@gunnaraasen I agree, but I suspect that database token auth might be handled by Cloud gateways for some products or deployments, i.e. not always by the router in IOx?
